test/e2e/basic_test.go

```diff
@@ -561,6 +509,86 @@ func testReconcileGatewayRoutesOnStartup(t *testing.T, data *TestData, isIPv6 bo
 	}
 }
 
+func TestCleanStaleClusterIPRoutes(t *testing.T) {
+	skipIfNumNodesLessThan(t, 2)
+	skipIfHasWindowsNodes(t)
+	skipIfProxyDisabled(t)
+
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+	skipIfProxyAllDisabled(t, data)
+
+	// Create a backend Pod for test Service: if a Service has no backend Pod, no ClusterIP route will be installed.
+	createAndWaitForPod(t, data, data.createNginxPodOnNode, "test-clean-stale-route-pod", nodeName(0), testNamespace, false)
+
+	if len(clusterInfo.podV4NetworkCIDR) != 0 {
+		t.Logf("Running IPv4 test")
+		testCleanStaleClusterIPRoutes(t, data, false)
+	}
+	if len(clusterInfo.podV6NetworkCIDR) != 0 {
+		t.Logf("Running IPv6 test")
+		testCleanStaleClusterIPRoutes(t, data, true)
+	}
+}
+
+func testCleanStaleClusterIPRoutes(t *testing.T, data *TestData, isIPv6 bool) {
+	ipProtocol := corev1.IPv4Protocol
+	if isIPv6 {
+		ipProtocol = corev1.IPv6Protocol
+	}
+	// Create two test ClusterIPs.
+	svc, err := data.createNginxClusterIPService(fmt.Sprintf("test-clean-stale-route-svc1-%v", isIPv6), testNamespace, false, &ipProtocol)
+	require.NoError(t, err)
+	require.NotEqual(t, "", svc.Spec.ClusterIP, "ClusterIP should not be empty")
+	svc, err = data.createNginxClusterIPService(fmt.Sprintf("test-clean-stale-route-svc2-%v", isIPv6), testNamespace, false, &ipProtocol)
+	require.NoError(t, err)
+	require.NotEqual(t, "", svc.Spec.ClusterIP, "ClusterIP should not be empty")
+	time.Sleep(time.Second)
+
+	nodeName := nodeName(0)
+	podName := getAntreaPodName(t, data, nodeName)
+	restartAntreaAgentPod(t, data, podName, nodeName)
```

Comment:
is there a reason why you don't use `deleteAntreaAgentOnNode` here? Unlike for flow tests, I think it should be ok to restart the Pod here, as opposed to just the container. Killing the container is not ideal as you need the deferred function below to reset the restart count. If you use `deleteAntreaAgentOnNode` you should not be needing that deferred function.

Reply:
You are right, just deleting the Antrea Agent is ok.